S3.Rmd

```diff
@@ -31,17 +31,18 @@ print(unclass(f))
 
 `unclass()` strips the class attribute from its input, so it is a useful tool for seeing what special behaviour an S3 class adds. 
 
-Becareful when using `str()`: some S3 classes provide a custom `str()` method which can hide the underlying details. For example, take the `POSIXlt` class, which is one of the two classes used to represent date-time data:
+`str()` shows the internal structure of S3 objects. Be careful when using `str()`: some S3 classes provide a custom `str()` method which can hide the underlying details. For example, take the `POSIXlt` class, which is one of the two classes used to represent date-time data:
 
 ```{r}
 time <- strptime("2017-01-01", "%Y-%m-%d")
 str(time)
 str(unclass(time), list.len = 5)
 ```
 
-Generics behave differently for different classes because generics have __methods__. A method is a function that implements the generic behaviour for a specific class. The generic doesn't actually do any work: its job is to find the right method and pass on its arguments. S3 methods are function with a special naming scheme, `generic.class()`. For example, the Date method for the `mean()` generic is called `mean.Date()`, and the factor method for `print()` is called `print.factor()`. This is the reason that most modern style guides discourage the use of `.` in function names: it makes them look like S3 methods. For example, is `t.test()` the `t` method for `test` objects? 
+A __generic__ and its __methods__ are functions that operate on classes. The role of a  generic is to find the right method for the for the arguments that it is provided; this is sometimes called __dispatch__.  A method is a function that implements the generic behaviour for a specific class. S3 methods are function with a special naming scheme, `generic.class()`. For example, the Date method for the `mean()` generic is called `mean.Date()`, and the factor method for `print()` is called `print.factor()`. This is the reason that most modern style guides discourage the use of `.` in function names: it makes them look like S3 methods. For example, is `t.test()` the `t` method for `test` objects?
 
-You can find some S3 methods (those in the base package and those that you've created) by typing their names. However, this will but will not work with most packages because S3 methods are not exported: they live only inside the package, and not available from the global environment. Instead, you can use `getS3method()`, which will work regardless of where the method lives:
+<!-- FIXME: it seems like getS3method() is old-school here -->
```

Comment:
Is there a better base R function? Or do you mean I should have a wrapper in S3?